From 47bcacfc1959724209542c1939cee16a9f41f082 Mon Sep 17 00:00:00 2001
From: Xianzhong <b07117@freescale.com>
Date: Thu, 12 Mar 2015 21:18:39 +0800
Subject: [PATCH] MGS-567 fix GPU kernel crash when setup mmu entries

the newer mmu requires the mapping size for the physical space,
mmu mapping size is up to 2G bytes according to gc2000+ mmu spec,

setting with 2G may waste some mmu entry memory, better with ddr size.

Signed-off-by: Xianzhong <b07117@freescale.com>
Acked-by: Jason Liu
---
 .../linux/kernel/platform/freescale/gc_hal_kernel_platform_imx6q14.c   | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_imx6q14.c b/drivers/mxc/gpu-viv/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_imx6q14.c
index 5e40678..bd0c6e1 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_imx6q14.c
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_imx6q14.c
@@ -412,6 +412,9 @@ gckPLATFORM_AdjustParam(
 
     Args->gpu3DMinClock = initgpu3DMinClock;
 
+    if (Args->physSize == 0)
+        Args->physSize = 0x80000000;
+
     return gcvSTATUS_OK;
 }
 
-- 
1.8.0

